src/main/network/http.ts

```diff
@@ -294,35 +294,27 @@ export const httpGetWithAuth =
             const options = _options || {};
             options.method = "get";
 
-            const response = await httpFetchFormattedResponse(
-                _url,
-                options,
-                enableAuth ? undefined : _callback,
-                ..._arg,
-            );
-
             if (enableAuth) {
-                if (response.statusCode === 401) {
-
-                    const url = _url instanceof URL ? _url : new URL(_url);
+                const url = _url instanceof URL ? _url : new URL(_url);
                     const { host } = url;
 
                     const auth = await getConfigRepoOpdsAuthenticationToken(host);
 
-                    if (
-                        typeof auth === "object"
-                        && auth.accessToken
-                    ) {
-                        return httpGetUnauthorized(auth)(_url, options, _callback, ..._arg);
-                    } else {
-                        debug("there are no authentication credentials in configRepo");
-                        return await handleCallback(response, _callback);
-                    }
-                } else {
-                    return await handleCallback(response, _callback);
+                if (
+                    typeof auth === "object"
+                    && auth.accessToken
+                ) {
+                    return httpGetUnauthorized(auth)(_url, options, _callback, ..._arg);
                 }
             }
-            return response;
+
+            return httpFetchFormattedResponse(
+                _url,
+                options,
+                enableAuth ? undefined : _callback,
+                ..._arg,
+            );
+
```

Comment:
This certainly needs reviewing carefully. This is taking us back to the original crude prototype which populated auth headers in HTTP requests before receiving any HTTP 401 responses.

Reply:
@saad-kh is there a link `rel` we can leverage (?) to discover that a given URL is an endpoint that necessitates auth, and which does not produce a 401 response in case of missing auth data.

Reply:
I suggest a call asap

Reply:
Hi Daniel, Pierre,

Here's the kind of link where we need to send an authentication token without waiting for a 401 response.

`<link href="" type="application/atom+xml;type=entry;profile=opds-catalog" rel="alternate">`

Reply:
Ah, so there is nothing "special" in `type` or `rel` that would allow us to reliably implement a workaround ... damn!